lib/debug_socket.rb

```diff
@@ -16,7 +17,10 @@ def self.logger
   end
 
   def self.start(path)
-    raise "debug socket thread already running" if @thread
+    pid = Process.pid
+    raise "debug socket thread already running for this process" if @thread && @pid == pid
```

Comment:
WAITAMINUTE `@pid = pid` is right below this, I completely missed that. changes LGTM :P